js/ui/data_grid.d.ts

```diff
@@ -69,767 +69,1019 @@ import Widget, {
+ */
+export interface DataErrorOcurredEvent<T extends GridBase> {
+  readonly component: T;
+  readonly element: TElement;
+  readonly model?: any;
+  readonly error: Error;
+}
+
+/**
+ * @public
+ */
+export interface EditCanceledEvent<T extends GridBase> {
+  readonly component: T;
+  readonly element: TElement;
+  readonly model?: any;
+  readonly changes: Array<any>;
+}
+
+/**
+ * @public
+ */
+export interface EditCancelingEvent<T extends GridBase> {
+  readonly component: T;
+  readonly element: TElement;
+  readonly model?: any;
+  readonly changes: Array<any>;
+  cancel: boolean;
+}
+
+
+/**
+ * @public
+ */
+export interface InitNewRowEvent<T extends GridBase> {
+  readonly component: T;
+  readonly element: TElement;
+  readonly model?: any;
+  data: any;
+  promise?: TPromise<void>;
+}
+
+/**
+ * @public
+ */
+export interface KeyDownEvent<T extends GridBase> {
+  readonly component: T;
+  readonly element: TElement;
+  readonly model?: any;
+  readonly event: TEvent;
+  handled: boolean;
+}
+
+/**
+ * @public
+ */
+export interface RowCollapsedEvent<T extends GridBase> {
+  readonly component: T;
+  readonly element: TElement;
+  readonly model?: any;
+  readonly key: any;
+}
+
+/**
+ * @public
+ */
+export interface RowCollapsingEvent<T extends GridBase> {
+  readonly component: T;
+  readonly element: TElement;
+  readonly model?: any;
+  readonly key: any;
+  cancel: boolean;
+}
+
+/**
+ * @public
+ */
+export interface RowExpandedEvent<T extends GridBase> {
+  readonly component: T;
+  readonly element: TElement;
+  readonly model?: any;
+  readonly key: any;
+}
+
+/**
+ * @public
+ */
+export interface RowExpandingEvent<T extends GridBase> {
+  readonly component: T;
+  readonly element: TElement;
+  readonly model?: any;
+  readonly key: any;
+  cancel: boolean;
+}
+
+/**
+ * @public
+ */
+export interface RowInsertedEvent<T extends GridBase> {
+  readonly component: T;
+  readonly element: TElement;
+  readonly model?: any;
+  readonly data: any;
+  readonly key: any;
+  readonly error: Error;
+}
+
+/**
+ * @public
+ */
+export interface RowInsertingEvent<T extends GridBase> {
+  readonly component: T;
+  readonly element: TElement;
+  readonly model?: any;
+  readonly data: any;
+  cancel: boolean | TPromise<void>;
+}
+
+/**
+ * @public
+ */
+export interface RowRemovedEvent<T extends GridBase> {
+  readonly component: T;
+  readonly element: TElement;
+  readonly model?: any;
+  readonly data: any;
+  readonly key: any;
+  readonly error: Error;
+}
+
+/**
+ * @public
+ */
+export interface RowRemovingEvent<T extends GridBase> {
+  readonly component: T;
+  readonly element: TElement;
+  readonly model?: any;
+  readonly data: any;
+  readonly key: any;
+  cancel: boolean | TPromise<void>;
+}
+
+/**
+ * @public
+ */
+export interface RowUpdatedEvent<T extends GridBase> {
+  readonly component: T;
+  readonly element: TElement;
+  readonly model?: any;
+  readonly data: any;
+  readonly key: any;
+  readonly error: Error;
```

Comment:
```suggestion
  ...
  readonly error?: Error;
```

error may not be defined